index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
-       *
-       * @returns Promise of the active public key
-       */
-      getActivePublicKey(): Promise<string>;
-
-      /**
-       * Add a token to ArConnect (if it is not already present)
-       *
-       * @param id Token contract ID
-       * @param type Type of the token (asset or collectible)
-       * @param gateway Gateway config for the token
-       */
-      addToken(id: string, type?: TokenType, gateway?: GatewayConfig): Promise<void>;
-
-      /**
-       * Checks if a token has been added to ArConnect
-       * 
-       * @param id Token to check for
-       * 
-       * @returns Token added or not
-       */
-      isTokenAdded(id: string): Promise<boolean>;
-
-      /**
-       * Dispatch an Arweave transaction (preferably bundled)
-       *
-       * @param transaction Transaction to dispatch
-       * 
-       * @returns Dispatched transaction ID and type
-       */
-      dispatch(transaction: Transaction): Promise<DispatchResult>;
-
-      /**
-       * Create a deterministic secret based on the input data.
-       * 
-       * @param data Input data to generate the hash from
-       * @param options Hash configuration
-       * 
-       * @returns Hash array
-       */
-      privateHash(data: ArrayBuffer, options: SignMessageOptions): Promise<Uint8Array>;
-
-      /**
-       * Create and sign a DataItem (bundled transaction item),
-       * that can be loaded into "arbundles".
-       * 
-       * @param dataItem Data item params
-       * 
-       * @returns Buffer of the signed data item
-       */
-      signDataItem(dataItem: DataItem): Promise<ArrayBufferLike>;
-
-      /**
-       * Create a cryptographic signature for any piece of data for later validation.
-       * This function cannot be used to sign transactions or interactions, because the data
-       * gets hashed before the signature generation.
-       * 
-       * @param data Message to be hashed and signed
-       * @param options Signature options
-       * 
-       * @returns Signature
-       */
-      signMessage(data: ArrayBuffer, options?: SignMessageOptions): Promise<Uint8Array>;
-
-      /**
-       * Verify a cryptographic signature created with the arweaveWallet.signMessage() function.
-       * 
-       * @param data Data to verify with the signature
-       * @param signature Signature to validate
-       * @param publicKey Optionally match the signature with a different public key than the currently active
-       * @param options Signature options
-       * 
-       * @returns Validity
-       */
-      verifyMessage(
-        data: ArrayBuffer,
-        signature: ArrayBuffer | string,
-        publicKey?: string,
-        options?: SignMessageOptions
-      ): Promise<boolean>;
-
-      /**
-       * Experimental event emitter. Allows listening to gateway config
-       * updates, bundler node changes, etc.
-       */
-      events: Emitter<InjectedEvents>;
-    };
+    arweaveWallet: ArweaveWallet;
```

Comment:
I exported the standalone `interface` below so that it can be used in other projects easily. In my case, I do `class Othent implements Omit<ArConnect, "connect">`.